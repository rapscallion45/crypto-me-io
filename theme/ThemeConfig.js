import PropTypes from 'prop-types';
import { useMemo } from 'react';
import { CacheProvider } from '@emotion/react';
import { CssBaseline } from '@mui/material';
import { ThemeProvider, createTheme } from '@mui/material/styles';
import shape from './shape';
import palette from './palette';
import typography from './typography';
import componentsOverride from './overrides';
import shadows, { customShadows } from './shadows';

const ThemeConfig = function Themeconfig({ emotionCache, children }) {
  const themeOptions = useMemo(
    () => ({
      palette,
      shape,
      typography,
      shadows,
      customShadows,
    }),
    []
  );

  const theme = createTheme(themeOptions);
  theme.components = componentsOverride(theme);

  return (
    <CacheProvider value={emotionCache}>
      <ThemeProvider theme={theme}>
        <CssBaseline />
        {children}
      </ThemeProvider>
    </CacheProvider>
  );
};

ThemeConfig.propTypes = {
  children: PropTypes.node,
};

export default ThemeConfig;
