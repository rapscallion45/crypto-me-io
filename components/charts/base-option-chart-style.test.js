import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import createEmotionCache from '../../utils/createEmotionCache';
import ThemeConfig from '../../theme/ThemeConfig';
import BaseOptionChartStyle from './base-option-chart-style';

describe('Charts Base Options', () => {
  it('Renders correctly enzyme', () => {
    /* Arrange */
    /* Act */
    const view = render(
      <ThemeConfig emotionCache={createEmotionCache()}>
        <BaseOptionChartStyle />
      </ThemeConfig>
    );

    /* Assert */
    expect(toJson(view)).toMatchSnapshot();
  });
});
