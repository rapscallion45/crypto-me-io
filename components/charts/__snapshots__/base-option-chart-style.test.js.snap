// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Charts Base Options Renders correctly enzyme 1`] = `null`;
